std/mutex.zig

```diff
@@ -122,7 +122,7 @@ else switch (builtin.os) {
     },
 };
 
-const TestContext = struct {
+const TestContext = packed struct {
```

Comment:
can you explain this change?

Reply:
This prevents a segfault, probably due to alignment, that otherwise happens here (according to GDB):
```
testing.expect(context.data == thread_count * TestContext.incr_count);
```

Reply:
I reverted this.